### PR TITLE
fix: Use size bucket for compacted segment size metric

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #29204

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -106,6 +106,9 @@ var (
 	// longTaskBuckets provides long task duration in milliseconds
 	longTaskBuckets = []float64{1, 100, 500, 1000, 5000, 10000, 20000, 50000, 100000, 250000, 500000, 1000000, 3600000, 5000000, 10000000} // unit milliseconds
 
+	// size provides size in MB
+	sizeBuckets = []float64{0.01, 0.1, 1, 100, 500, 1024, 2048, 4096, 10000, 50000} // unit MB
```
Comment (LoveEachDay) [inline pkg/metrics/metrics.go:110]: We should keep the size in base unit, like Byte for storage. 
Here the bucket should begin with 10000 Byte.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**